### PR TITLE
Health auth screens

### PR DESCRIPTION
**Pull request by crugwiro**
## Changes
- Implements screens that allow a mobile user to control health data permissions.
- Prompts users to enable the authorization

## Requirements
N/A

## PR Dependencies
N/A

## Merged Branches and Other Dependencies
Merged branch(es):
Other dependencies:

## Subtasks / Progress
TODO

## Preview
TODO

## Testing
Manual 
For testing:
- Make sure DOB is set in Apple healthKit

### CODE REVIEW

**Comment by wynnset** [comment]
Hi @crugwiro you had mentioned this would be ready by Wednesday so I just wanted to follow up. Thank you

**Comment by crugwiro** [comment]
@wynnset should be ready by end of day today.

**Comment by crugwiro** [comment]
@wynnset This is now ready for review, assigning it to @cara-wong

**Review by cara-wong — CHANGES_REQUESTED** [review]
Tested this PR on a real ios device, an ios sim and an android sim. Found some user flow issues - working with Clement to point out the issues.

File: mobile/src/controllers/Notifications.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { ILocalSettingsController } from './LocalSettings';
 import { ScheduleResult } from 'common/models/Notifications';
 import { ThrottleAction } from 'common/utils/throttle';
 import { IDisposable } from 'common/utils/unsubscriber';
+//
```
Comment (cara-wong) [inline mobile/src/controllers/Notifications.ts:10]: Unneeded cosmetic changes in this file (start of comment, removed line)

File: mobile/src/controllers/User.ts [hunk 2]
```diff
@@ -244,6 +254,8 @@ export class UserController extends UserControllerBase implements IUserControlle
             }
 
             await this._localSettings.load(user.id);
+            // await this.notifications.initAsync();
```
Comment (cara-wong) [inline mobile/src/controllers/User.ts:257]: is this comment needed?

File: mobile/src/controllers/index.ts [hunk 3]
```diff
@@ -97,6 +97,9 @@ export default class AppController implements IAppController {
         this.version.checkForUpdates();
 
         if (this.User && this.User.user) {
+            logger.log('INITIALIZE PERM/HEALTH: INIT HEALTH');
+            // await this.User.notifications.initAsync();
```
Comment (cara-wong) [inline mobile/src/controllers/index.ts:101]: is this comment needed?

File: mobile/src/helpers/health.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  permissions: {
+    read: ["StepCount", "BiologicalSex", "DateOfBirth"],
+    write: [],
+  },
+};
+
+export const auth = async() => {
+  var isAuth;
+
+  return await GoogleFit.checkIsAuthorized().then (() => {
```
Comment (cara-wong) [inline mobile/src/helpers/health.ts:23]: are the comments in this method needed? They look like old logic that is no longer being used

File: mobile/src/stateMachine/scenario.ts [hunk 5]
```diff
@@ -71,6 +74,30 @@ export const MasloScenario: GlobalScenario<States> = {
...
             { target: States.SignInWithEmail, trigger: Triggers.Secondary },
+            { target: States.HealthConsent, trigger: Triggers.Primary },
+        ],
+    },
+    [States.HealthConsent]: {
+        view: HealthConsentView,
+        exit: [
+            { target: States.HealthScopes, trigger: Triggers.Primary },
+            { target: States.HomeRouter, trigger: Triggers.Submit },
+            // { target: States.HomeRouter, trigger: Triggers.Submit },
```
Comment (cara-wong) [inline mobile/src/stateMachine/scenario.ts:85]: comment needed?

File: mobile/src/stateMachine/scenario.ts [hunk 6]
```diff
@@ -332,7 +362,8 @@ export const MasloScenario: GlobalScenario<States> = {
             { target: States.Profile, trigger: Triggers.Back },
             { target: States.NotificationsSettings, trigger: Triggers.Primary },
             { target: States.ChangePassword, trigger: Triggers.Submit },
-            { target: States.ConfirmAccount, trigger: Triggers.Secondary },
+            // { target: States.ConfirmAccount, trigger: Triggers.Secondary },
```
Comment (cara-wong) [inline mobile/src/stateMachine/scenario.ts:365]: comment needed?

File: mobile/src/stateMachine/views/healthData/healthScopes.tsx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+import { HealthPermissionsViewModel } from 'src/viewModels/HealthPermissionsViewModel';
+import TextStyles from 'src/styles/TextStyles';
+import Dots from 'src/components/Dots';
+
+
+@observer
+export class HealthScopesView extends ViewState {
+    constructor(props) {
+        super(props);
+        // this._contentHeight = this.persona.setupContainerHeight(minContentHeight, { rotation: 405 });
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/healthData/healthScopes.tsx:23]: comment needed?

File: mobile/src/stateMachine/views/healthData/healthScopes.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+        this._contentHeight = this.persona.setupContainerHeightForceScroll();
+    }
+
+    state = {
+        // opacity: new Animated.Value(0),
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/healthData/healthScopes.tsx:28]: comment needed?

File: mobile/src/stateMachine/views/login/notificationsPermission.ts [hunk 9]
```diff
@@ -2,6 +2,7 @@ import { ViewState } from '../base';
 import AppController from 'src/controllers';
 import { observer } from 'mobx-react';
 import { ScenarioTriggers } from '../../abstractions';
+import GoogleFit, { Scopes } from 'react-native-google-fit';
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/login/notificationsPermission.ts:5]: unused import statement - is this needed?

File: mobile/src/stateMachine/views/main/home.tsx [hunk 10]
```diff
@@ -279,15 +312,18 @@ export class HomeView extends ViewState<{ opacity: Animated.Value }> {
         const {
             loading,
         } = this.viewModel;
-
+        const healthPerm = Platform.OS == 'ios'? this.originalIsEnabledOG : this.originalIsEnabled
         return (
             <MasloPage style={[this.baseStyles.page, { backgroundColor: Colors.home.bg }]}>
                 <Animated.View style={[this.baseStyles.container, styles.container, { height: this._contentHeight, opacity: this.state.opacity }]}>
+                    {/* <ScrollView> */}
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/main/home.tsx:319]: comments in the return statement needed?

File: mobile/src/stateMachine/views/main/home.tsx [hunk 11]
```diff
@@ -338,4 +374,24 @@ const styles = StyleSheet.create({
...
+        height: 30,
+        borderColor: Colors.welcome.mailButton.border,
+        borderWidth: 1,
+        justifyContent: 'center'
+    },
+    mailButtonTitle: {
+        color: Colors.welcome.mailButton.title,
+    },
+    healthView: {
+        // height: 100,
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/main/home.tsx:388]: comment needed?

File: mobile/src/stateMachine/views/healthData/permInstructions.tsx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+import { observer } from 'mobx-react';
+import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native';
+import { MasloPage, Container } from 'src/components';
+import { ScenarioTriggers } from '../../abstractions';
+import { createLogger } from 'common/logger';
+import { styles } from 'react-native-markdown-renderer';
+import Images from 'src/constants/images';
+import Dots from 'src/components/Dots';
+
+export const logger = createLogger('[endQOL]');
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/healthData/permInstructions.tsx:12]: logger shouldn't preceed logs with "endQol" for this view

File: mobile/src/stateMachine/views/healthData/permInstructions.tsx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+const minContentHeight = 1000;
+
+const display = [{title : 'Open the Apple Health App on your phone',image: Images.screen2},
+{title : 'Next, open your personal health settings by tapping the top right corner in the App ', image: Images.screen2},
```
Comment (cara-wong) [inline mobile/src/stateMachine/views/healthData/permInstructions.tsx:17]: App here shouldn't be capitalized

File: mobile/src/viewModels/HealthPermissionsViewModel.ts [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+import * as Links from 'src/constants/links';
+import Images from 'src/constants/images';
+
+const logger = createLogger('[HealthPermissionsViewModel]');
+
+// this can be updated when we want to add/remove permissions
+const permissions = [{title:"Sex", icon: Images.difficultIcon, description: "Allow Companion kit to access sex info"}
+,{title:"Date Of Birth", icon: Images.veryPositiveIcon, description: "Allow Companion kit to access DOB info"},
+ {title:"Active Steps", icon:Images.busIcon, description: "Allow Companion kit to access Steps info"},
+{title:"BMI", icon:Images.keyIcon, description: "Allow Companion kit to access BMI info"}];
```
Comment (cara-wong) [inline mobile/src/viewModels/HealthPermissionsViewModel.ts:14]: should we be showing BMI if we don't ask for it?

**Review by cara-wong — CHANGES_REQUESTED** [review]
This is looking really good!
I left some comments on certain lines. Some of the comments are regarding what seems like old logic that is just commented out, so I'm checking if we can remove these. Some comments are also just questions, so feel free to resolve if it's a non-issue. In addition:
1. On the Health Data page in settings, in the design the text reading "We are collecting your health data to build..." is on this page whether the permissions are enabled or disabled. Right now there's only text when it's disabled.
2. On a real ios device, the settings link takes you to the wrong place in settings. I know you mentioned this last meeting, so just documenting it so it can be fixed before this PR is merged.
3. On my iphone 8+, when permissions are disabled and there's a prompt on the home screen, the past check-ins are slightly cut off. (suggests we should be testing this on multiple screen sizes to ensure dynamic UI)
4. Probably good to have someone else test this on a real android device

**Comment by rlnsy** [comment]
@SukiCai just a reminder to take care of this PR when you have a chance (make sure the comments are resolved, make sure it is fully merged into your branch, and close when done)

**Comment by SukiCai** [comment]
Hi @rlnsy, I've resolved all the comment above and fully merged into PR #52 , we can close this PR now

**Pull request closed without merging**